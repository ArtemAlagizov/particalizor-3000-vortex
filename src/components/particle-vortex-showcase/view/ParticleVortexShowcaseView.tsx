import React from "react";
import { IParticleVortexShowcaseView } from "./IParticleVortexShowcaseView";
import { ParticleVortex } from "particalizor-3000";
import { Styled } from "../styled";

const ParticleVortexShowcaseView: React.FC<IParticleVortexShowcaseView> = ({
  particleNumberValue,
  particleTraceWidthValue,
  particleLifeTimeValue,
  imageWidth,
  imageHeight,
  backgroundColor,
  vortexNumberValue
}) => {
  return (
    <Styled.ShowcaseComponentContainer>
      <ParticleVortex
        imageWidth={imageWidth}
        imageHeight={imageHeight}
        vortexNumber={vortexNumberValue}
        particleTraceWidth={particleTraceWidthValue}
        particleNumber={particleNumberValue}
        particleLifeTime={particleLifeTimeValue}
        backgroundColor={backgroundColor}
      />
    </Styled.ShowcaseComponentContainer>
  );
};

export default ParticleVortexShowcaseView;
