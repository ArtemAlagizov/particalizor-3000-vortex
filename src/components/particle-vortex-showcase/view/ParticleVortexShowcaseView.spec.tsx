import React from "react";
import { shallow } from "enzyme";
import ParticleVortexShowcaseView from "./ParticleVortexShowcaseView";

describe("Given a ParticleVortexShowcaseView", () => {
  describe("when rendering", () => {
    it("should match snapshot", () => {
      const component = shallow(
        <ParticleVortexShowcaseView
          particleTraceWidthValue={1}
          particleNumberValue={1}
          particleLifeTimeValue={1}
          backgroundColor={"red"}
          vortexNumberValue={3}
        />
      );

      expect(component).toMatchSnapshot();
    });
  });
});
