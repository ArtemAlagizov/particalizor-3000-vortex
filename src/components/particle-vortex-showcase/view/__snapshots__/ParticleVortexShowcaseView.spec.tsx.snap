// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a ParticleVortexShowcaseView when rendering should match snapshot 1`] = `ShallowWrapper {}`;
